.changelog/12573.txt

```diff
@@ -0,0 +1,4 @@
+```release-note:feature
+rpc: (beta-only): add middleware to net/rpc calls and record a new metric `consul.rpc.server.call` with labels
+for `method`, `errored`, `rpc_type`, `request_type`.
+```
```

Comment:
```suggestion
```release-note:feature
rpc: (beta-only): add a new metric `consul.rpc.server.call` with labels
for `method`, `errored`, `rpc_type`, `request_type`.
```
```

I think we should also add a separate changelog for switching to a net/rpc fork.

Reply:
I can do that in a separate PR that points to https://github.com/hashicorp/consul/pull/12311